crates/iceberg/src/spec/manifest/mod.rs

```diff
@@ -1056,4 +1091,120 @@ mod tests {
         assert!(!partitions[2].clone().contains_null);
         assert_eq!(partitions[2].clone().contains_nan, Some(false));
     }
+
```

Comment:
As discussed [here](https://github.com/apache/iceberg-rust/pull/1511/files#r2222019640), I'm planning to use snapshot test instead